File: include/PolyWorksSDK/samples/plugins/common/CSharp/CSharpCommands/COM_SDK/Util/CComCommandParameterInfoImpl.cs
// =================================================================================================
// Implementation of the class describing a COM command parameter
//
// Copyright © InnovMetric Logiciels Inc. 2009 All Rights Reserved
// -------------------------------------------------------------------------------------------------
namespace IM
{
    // =================================================================================
    // COM command parameter class description implementation
    // A description, a direction and a type must at least be specified
    // Otherwise, only needed members require initialisation
    // ---------------------------------------------------------------------------------
    public class CComCommandParameterInfoImpl
    {
        // =============================================================================
        // Constructor
        //
        // Parameter: desc_ : Description
        // Parameter: dir_  : Direction
        // Parameter: type_ : Type
        // -----------------------------------------------------------------------------
        public CComCommandParameterInfoImpl(string desc_, 
                                            IMPluginLib.EParameterDirection dir_, 
                                            IMPluginLib.EParameterType type_)
        {
            m_desc = desc_;
            m_dir  = dir_;
            m_type = type_;
        }

        // =============================================================================
        // Copy constructor
        //
        // Parameter: info_ : Object to be copied
        // -----------------------------------------------------------------------------
        public CComCommandParameterInfoImpl(CComCommandParameterInfoImpl info_)
        {
            if (info_ != null)
            {
                m_desc                  = info_.m_desc;
                m_values                = info_.m_values;
                m_dir                   = info_.m_dir;
                m_type                  = info_.m_type;
                m_hasDoubleDefaultValue = info_.m_hasDoubleDefaultValue;
                m_doubleDefaultValue    = info_.m_doubleDefaultValue;
                m_hasIntDefaultValue    = info_.m_hasIntDefaultValue;
                m_intDefaultValue       = info_.m_intDefaultValue;
                m_hasStringDefaultValue = info_.m_hasStringDefaultValue;
                m_stringDefaultValue    = info_.m_stringDefaultValue;
            }
        }

        public string                          m_desc;                              // Description 
        public string                          m_values = string.Empty;             // Possible values (in a string, separated by comas)
        public IMPluginLib.EParameterDirection m_dir;                               // Direction
        public IMPluginLib.EParameterType      m_type;                              // Type
        public bool                            m_hasDoubleDefaultValue;             // True if parameter have a default double value
        public double                          m_doubleDefaultValue;                // Default value for double parameter
        public bool                            m_hasIntDefaultValue;                // True if parameter have a default integer value
        public int                             m_intDefaultValue;                   // Default value for integer parameter
        public bool                            m_hasStringDefaultValue;             // True if parameter have a default string value
        public string                          m_stringDefaultValue = string.Empty; // Default value for string parameter
    }
}

File: include/PolyWorksSDK/samples/plugins/common/VisualBasic.NET/VBNet3DSceneRendering/COM_SDK/Util/CComCommandParameterInfoImpl.vb
' =================================================================================================
' Implementation of the class describing a COM command parameter
'
' Copyright © InnovMetric Logiciels Inc. 2012 All Rights Reserved
' -------------------------------------------------------------------------------------------------
Namespace VBNet3DSceneRendering

    ' =================================================================================
    ' COM command parameter class description implementation
    ' A description, a direction and a type must at least be specified
    ' Otherwise, only needed members require initialisation
    ' ---------------------------------------------------------------------------------
    Public Class CComCommandParameterInfoImpl

        ' =============================================================================
        ' Constructor
        '
        ' Parameter: desc_ : Description
        ' Parameter: dir_  : Direction
        ' Parameter: type_ : Type
        ' -----------------------------------------------------------------------------
        Public Sub New(ByVal desc_ As String, _
                       ByVal dir_ As IMPluginLib.EParameterDirection, _
                       ByVal type_ As IMPluginLib.EParameterType)

            m_desc = desc_
            m_dir = dir_
            m_type = type_

        End Sub

        ' =============================================================================
        ' Copy constructor
        '
        ' Parameter: info_ : Object to be copied
        ' -----------------------------------------------------------------------------
        Public Sub New(ByVal info_ As CComCommandParameterInfoImpl)

            If Not info_ Is Nothing Then

                m_desc = info_.m_desc
                m_values = info_.m_values
                m_dir = info_.m_dir
                m_type = info_.m_type
                m_hasDoubleDefaultValue = info_.m_hasDoubleDefaultValue
                m_doubleDefaultValue = info_.m_doubleDefaultValue
                m_hasIntDefaultValue = info_.m_hasIntDefaultValue
                m_intDefaultValue = info_.m_intDefaultValue
                m_hasStringDefaultValue = info_.m_hasStringDefaultValue
                m_stringDefaultValue = info_.m_stringDefaultValue

            End If

        End Sub

        Public m_desc As String                               ' Description 
        Public m_values As String = String.Empty              ' Possible values (in a string, separated by comas)
        Public m_dir As IMPluginLib.EParameterDirection       ' Direction
        Public m_type As IMPluginLib.EParameterType           ' Type
        Public m_hasDoubleDefaultValue As Boolean             ' True if parameter have a default double value
        Public m_doubleDefaultValue As Double                 ' Default value for double parameter
        Public m_hasIntDefaultValue As Boolean                ' True if parameter have a default integer value
        Public m_intDefaultValue As Integer                   ' Default value for integer parameter
        Public m_hasStringDefaultValue As Boolean             ' True if parameter have a default string value
        Public m_stringDefaultValue As String = String.Empty  ' Default value for string parameter

    End Class

End Namespace



